sdk/samples/sample_common.c

```diff
@@ -0,0 +1,218 @@
+            case AUDIO_PTIME_333US:
+                slices_per_second = 3003;
+                break;
+            default:
+                slices_per_second = 1;
+                break;
+            }
+        /* no break is intentional here */
+        case AUDIO_SAMPLING_44K:
+            switch (ptime) {
+            case AUDIO_PTIME_1MS:
+            case AUDIO_PTIME_1_09MS: // 44100 Hz
+                slices_per_second = 1000;
+                break;
+            case AUDIO_PTIME_125US:
+            case AUDIO_PTIME_0_14MS: // 44100 Hz
+                slices_per_second = 8000;
+                break;
+            case AUDIO_PTIME_80US:
+            case AUDIO_PTIME_0_09MS: // 44100 Hz
+                slices_per_second = 12500;
+                break;
+            default:
+                slices_per_second = 1;
+                break;
+            }
+            break;
+        default:
+            slices_per_second = 1;
+            break;
+    }
+
+    if (sampling == AUDIO_SAMPLING_96K){
+        samples_per_second = 96000;
+    } else if (sampling == AUDIO_SAMPLING_44K){
+        samples_per_second = 44100;
+    } else { // if (sampling == AUDIO_SAMPLING_48K)
+        samples_per_second = 48000;
+    }
+
+    return audio_channels * bits_per_sample * samples_per_second * 1.0 / slices_per_second; // bits per slice
+}
+
+static int getFrameSize(video_pixel_format fmt, uint32_t width, uint32_t height, bool interlaced)
+{
+    size_t size = 0;
+    size_t pixels = (size_t)(width*height);
+    switch (fmt) {
+        /* YUV 422 packed 8bit
+        (aka ST20_FMT_YUV_422_8BIT,
+        aka ST_FRAME_FMT_UYVY) */
+        // TODO: Ensure it was supposed to be packed, not planar
+        case PIX_FMT_YUV422P:
+            size = pixels * 2;
+            break;
+        /* 8 bits RGB pixel in a 24 bits
+        (aka ST_FRAME_FMT_RGB8) */
+        case PIX_FMT_RGB8:
+            size = pixels * 3;
+            break;
+/* Customized YUV 420 8bit, set transport format as ST20_FMT_YUV_420_8BIT. For direct transport of
+none-RFC4175 formats like I420/NV12. When this input/output format is set, the frame is identical to
+transport frame without conversion. The frame should not have lines padding) */
+        /* PIX_FMT_NV12, YUV 420 planar 8bits
+        (aka ST_FRAME_FMT_YUV420CUSTOM8,
+        aka ST_FRAME_FMT_YUV420PLANAR8) */
+        case PIX_FMT_NV12:
+            size = pixels * 3 / 2;
+            break;
+        case PIX_FMT_YUV444P_10BIT_LE:
+            size = pixels * 2 * 3;
+            break;
+        /* YUV 422 planar 10bits little indian, in two bytes
+        (aka ST_FRAME_FMT_YUV422PLANAR10LE) */
+        case PIX_FMT_YUV422P_10BIT_LE:
+        default:
+            size = pixels * 2 * 2;
+    }
+    if (interlaced) size /= 2; // TODO: Check if all formats support interlace
+    return (int)size;
+}
+
+/* print a description of all supported options */
+static void usage(FILE* fp, const char* path, int is_sender)
```

Comment:
I don't agree with extracting this from sender_app and recver_app and putting here with the `is_sender` argument to differentiate the output according to where it was called: from sender_app or recver_app. This is overengineering and premature optimization at the same time.

Please move the `usage()` implementation back to sender_app and recver_app respectively.

Reply:
Sorry, but I politely disagree. This function is used just for help message with usage, and does nothing else. This message has 20 switches for both apps, 3 for just sender, 1 for just receiver (with a comment in line 174 for clarity). So differing by just a few switches, 4/24 ~= 16%, to be exact.

In my opinion, duplicating that in 2 files will open the doors to misalign the switches between sender and receiver, and double the work when extending/changing/removing anything.

If your suggestion is to change the name of the variable, please provide a proposed name.

Reply:
Mateusz, I'm not insisting on cancelling that change. You have your strong arguments to keep that this way – this is fine, I agree.

Reply:
@Mionsz What do you think?